pkg/controller/apikey/create_test.go

```diff
@@ -43,51 +45,90 @@ func TestHandleCreate(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Create a browser runner.
-	browserCtx := browser.New(t)
-	taskCtx, done := context.WithTimeout(browserCtx, 30*time.Second)
-	defer done()
-
-	var apiKey string
-	if err := chromedp.Run(taskCtx,
-		// Pre-authenticate the user.
-		browser.SetCookie(cookie),
-
-		// Visit /apikeys/new.
-		chromedp.Navigate(`http://`+harness.Server.Addr()+`/realm/apikeys/new`),
-
-		// Wait for render.
-		chromedp.WaitVisible(`body#apikeys-new`, chromedp.ByQuery),
-
-		// Fill out the form.
-		chromedp.SetValue(`input#name`, "Example API key", chromedp.ByQuery),
-		chromedp.SetValue(`select#type`, strconv.Itoa(int(database.APIKeyTypeDevice)), chromedp.ByQuery),
-
-		// Click the submit button.
-		chromedp.Click(`#submit`, chromedp.ByQuery),
-
-		// Wait for the page to reload.
-		chromedp.WaitVisible(`body#apikeys-show`, chromedp.ByQuery),
-
-		// Get the API key.
-		chromedp.Value(`#apikey-value`, &apiKey, chromedp.ByQuery),
-	); err != nil {
-		t.Fatal(err)
-	}
-
-	// Ensure API key is valid.
-	record, err := harness.Database.FindAuthorizedAppByAPIKey(apiKey)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if got, want := record.RealmID, realm.ID; got != want {
-		t.Errorf("expected %v to be %v", got, want)
-	}
-	if got, want := record.Name, "Example API key"; got != want {
-		t.Errorf("expected %v to be %v", got, want)
-	}
-	if got, want := record.APIKeyType, database.APIKeyTypeDevice; got != want {
-		t.Errorf("expected %v to be %v", got, want)
-	}
+	t.Run("middleware", func(t *testing.T) {
+		t.Parallel()
```

Comment:
Seeing test races. chromedp might not be able to handle as much parallel

Reply:
Nah, it's the renderer. There's a legit data race that occurs in debug mode when templates are reloaded. We'd never hit it in prod because templates aren't reloaded. Fixed.